assets/src/hooks/useNotifications.ts

```diff
@@ -0,0 +1,52 @@
+import { Channel, Socket } from "phoenix"
+import { useContext, useEffect } from "react"
+import { SocketContext } from "../contexts/socketContext"
+import { StateDispatchContext } from "../contexts/stateDispatchContext"
+import { reload } from "../models/browser"
+import { notificationsFromData } from "../models/notificationData"
+import { Notification } from "../realtime.d"
+
+export const useNotifications = (
+  handleNewNotification: (notification: Notification) => void
+): void => {
+  const { socket }: { socket: Socket | undefined } = useContext(SocketContext)
+  const topic: string = "notifications"
+  const event: string = "notifications"
+  const [{ selectedRouteIds }] = useContext(StateDispatchContext)
+
+  useEffect(() => {
+    let channel: Channel | undefined
+
+    if (socket !== undefined) {
```

Comment:
@skyqrose I'm assuming that you're rolling a fresh implementation of this because the existing `useChannel` isn't suitable for some reason?

Reply:
Yeah. `useChannel` is more meant for "I'll return the most up-to-date state of the world". It keeps its own state under the hood. For notifications, the caller keeps track of the state, and the channel pushes the new notifications incrementally through a callback. (The notifications component needs to keep the state so it can remove read cards.)

Also because of this difference, the notifications channel doesn't return anything on join which would break `useChannel` when it tries to parse the response.

If you think it's worth it, we could have a single channel implementation like this:
```ts
useChannelCallback = (socket, topic, event, callback) => {
  useEffect(() => {
    // only implementation of channel.join
    channel.on(event, callback)
  })
}

useChannel = (socket, topic, event, parser) => {
  [state, setState] = useState()
  useChannelCallback(socket, topic, event, parse and setState)
  return state
}

useNotifications = (callback) => {
  useChannelCallback(socket, "notifications", "notifications", parse and callback)
}
```
but we'd have to either return an empty list (instead of returning nothing) when joining the notification channel, or let `useChannelCallback` decide whether to try to parse the join response.